core/src/main/java/com/gdx/game/common/UtilityClass.java

```diff
@@ -0,0 +1,49 @@
+package com.gdx.game.common;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+public class UtilityClass {
+
+
+    public static <T, E> Set<T> getKeysByValue(Map<T, E> map, E value) {
+        Set<T> keys = new HashSet<>();
+        for (Map.Entry<T, E> entry : map.entrySet()) {
+            if (Objects.equals(value, entry.getValue())) {
+                keys.add(entry.getKey());
+            }
+        }
+        return keys;
+    }
+
+    public static <T, E> Optional<T> getFirstKeyByValue(Map<T, E> map, E value){
+        return getKeysByValue(map, value).stream().findFirst();
+    }
+
+    public static <T, E> HashMap<E, T> mapInverter(HashMap<T, E> hashMap){
```

Comment:
you don't need this one if you use the mapInverter line 38

Reply:
Technically this hashmap is intended has Guava BiMap. For not implementing externally libraries I used this approach where the control map is built <Integer, String> where in the integer key there are the key code and in the string value the action associated. Unfortunately the from json method transform the map in a map of strings,  so to find the action associated at the key code I invert the map but after find the action I have to restore the previous order of the map so when you press back button and save the map as json you have the correct order 😁

Reply:
I've miss the point, sorry. I've used the invert map instead of the find by value because in this case is not necessary to remove the previous entry. For example if I change one key with the find from value an I make a put I'll have, for example, two options binding so I've to remove the found value before the put. If I use the invert instead, the option become the key so if I perform a put the hash map simply overwrite the entry 😁